### ISSUE TITLE
`.unionAll()` producing double `LIMIT`

### ISSUE THREAD

**Issue by spiffytech** [kysely-org/kysely#1304]
I'm writing a query that takes the first matching row from each of two subselects:

```ts
const select = db
  .selectFrom("links")
  .where(sql.val(true))
  .selectAll()
  .limit(1);

const rows = await select
  .unionAll(select)
  .execute();
```

_(Of course, I've simplified the query here.)_

Kysely turns this into invalid SQL:

```sql
SELECT
  *
FROM
  "links"
WHERE
  $1
UNION ALL
SELECT
  *
FROM
  "links"
WHERE
  $2
LIMIT
  $3
LIMIT
  $4
```

Postgres doesn't like the duplicated `LIMIT`. My intent was for each subselect to have a `LIMIT` applied, like so:

```sql
(
  SELECT
    *
  FROM
    "links"
  WHERE
    $1
  LIMIT
    $2
)
UNION ALL
(
  SELECT
    *
  FROM
    "links"
  WHERE
    $3
  LIMIT
    $4
)
```

As a workaround, I can do this:

```ts
(await db.executeQuery(sql`${select} union all ${select}`.compile(db))).rows
```

**Comment by koskimas** [comment]
Yeah, set operations are weird. Each dialect has implemented them differently. On some dialects, the unioned set is a part of the parent query and some parts of the parent query apply to the whole union (like limit here). On some dialects set operations can act like a set of individual subqueries where no query is the "main" query if you insert parentheses correctly, but this doesn't work on all dialects!

The only sane way to fix this really is to add a new free `union` function (and also `intersect`, `except` and so on). Something like this

```ts
unionAll([
  select,
  select
])
```

**Comment by spiffytech** [comment]
Would your proposed free `unionAll` function still compose with other queries?

```ts
db
  // Whatever's appropriate here
  .selectFrom(unionAll([...]))
  .where('foo', '=', 'bar')
  .selectAll()
  .execute()
```

---

Would it make sense to put the array variant of `unionAll` on the `db` object? That avoids extra imports, though it muddies which SQL semantics apply to any given `unionAll` call.

```ts
db.unionAll([...]).execute()
```

---

Would I be correct that Kysely should never generate the SQL `LIMIT $1 LIMIT $2`, regardless of dialect or intended parentheses? I don't think that's valid syntax for any DB.
